Products/DataCollector/ApplyDataMap/incrementalupdate.py

```diff
@@ -123,6 +124,20 @@ def parent(self):
 
         return self._parent
 
+    @property
+    def relname(self):
+        try:
+            if self.__relname is _NOTSET:
+                self.__relname = self.path.split('/')[-2]
```

Comment:
I'm not sure if there's really a need / benefit to trying to invent a relname and a parent for an objectmap that only has a compname.   The compname already gives you the target, and that's really all you should need?

Certainly that's how the old version worked..  Its code is very simple:

```
        if hasattr(datamap, "compname"):
            if datamap.compname:
                try:
                    tobj = device.getObjByPath(datamap.compname)
                except NotFound:
                    log.warn("Unable to find compname '%s'" % datamap.compname)
                    return False
            else:
                tobj = device
```

Could we just do something like that in target(), and let the relname be None?

Reply:
the relname is needed to get the relationship object, and for removing objects from relationships.

If we modify the target() property to as suggested, it means that finding the target will require a compname (path) to be specified, and datamaps that instead specify the relname and target_id would fail, or have to construct a compname from those attributes.  I think its better to extract simpler attributes from the compound compname instead.